content-src/components/Spotlight/Spotlight.js

```diff
@@ -129,13 +146,24 @@ const Spotlight = React.createClass({
     return (<section className="spotlight">
       <h3 className="section-title">Highlights <Hint id="highlights_hint" title="Highlights" body={HIGHLIGHTS_HINT_TEXT} /></h3>
       <ul className="spotlight-list">
-        {sites.map((site, i) => <SpotlightItem
-          index={i}
-          key={site.guid || site.cache_key || i}
-          page={this.props.page}
-          source="FEATURED"
-          onClick={this.onClickFactory(i, site)}
-          {...site} />)}
+        {sites.map((site, i) => {
+          if (this.props.placeholder) {
```

Comment:
Good question!

TopSites has a DEFAULT_LENGTH property of 6, but that's really just the maximum length, as its used to slice out of the sites array.

Spotlights has a default "length" property of 3, but NewTabPage overrides it with HIGHLIGHTS_LENGTH from common/constants.js.  Again, this is really a maximum length, as it's used to slice out of "sites".

console logging shows the things being passed from the selectors to NewTabs as 6 and 3, which I suspect is just our default set of TopSites and the three (soon-to-disappear) Highlights.

Which means that I suspect you're onto something here, and that at least for the Highlights, we should do something to ensure that we have some skeletons all the time.